### PR TITLE
spelling: fix spelling typo premption -> preemption

### PR DESCRIPTION
**Pull request by anchao**
## Summary

spelling: fix spelling typo premption -> preemption
spelling: fix spelling typo premp -> preemp

Signed-off-by: chao an <anchao@lixiang.com>

## Impact

N/A

## Testing

ci-check

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



No. This PR summary lacks crucial information.  While it mentions the spelling changes, it doesn't explain *where* these changes occur (what functional part of the code).  It also lacks context: *why* were these typos present?  Were they in comments, variable names, function names, etc.?  This context helps reviewers understand the scope and potential ramifications of even seemingly trivial changes.

The "Impact" section being simply "N/A" is insufficient.  At minimum, it should explicitly state "NO" for each impact category to indicate that the author considered those aspects.

The "Testing" section is also inadequate. "ci-check" doesn't provide any actual testing results.  The template specifically requests "Testing logs before change" and "Testing logs after change".  Even for a simple spelling fix, demonstrating that the code still compiles and (ideally) functions as expected is good practice.  Information on the build host and target is also missing.

**Review by lupyuen — APPROVED** [review]

**Review by acassis — APPROVED** [review]

File: drivers/note/notelog_driver.c [hunk 1]
```diff
@@ -61,7 +61,7 @@ static void notelog_cpu_resumed(FAR struct note_driver_s *drv,
 #  endif
 #endif
 #ifdef CONFIG_SCHED_INSTRUMENTATION_PREEMPTION
-static void notelog_premption(FAR struct note_driver_s *drv,
+static void notelog_preemption(FAR struct note_driver_s *drv,
                               FAR struct tcb_s *tcb, bool locked);
```
Comment (xiaoxiang781216) [inline drivers/note/notelog_driver.c:65]:
```suggestion
                               FAR struct tcb_s *tcb, bool locked);
```

File: drivers/note/notelog_driver.c [hunk 2]
```diff
@@ -103,7 +103,7 @@ static const struct note_driver_ops_s g_notelog_ops =
 #  endif
 #endif
 #ifdef CONFIG_SCHED_INSTRUMENTATION_PREEMPTION
-  notelog_premption,     /* premption */
+  notelog_preemption,     /* preemption */
```
Comment (xiaoxiang781216) [inline drivers/note/notelog_driver.c:106]:
```suggestion
  notelog_preemption,    /* preemption */
```

File: drivers/note/notelog_driver.c [hunk 3]
```diff
@@ -255,7 +255,7 @@ static void notelog_cpu_resumed(FAR struct note_driver_s *drv,
 #endif
 
 #ifdef CONFIG_SCHED_INSTRUMENTATION_PREEMPTION
-static void notelog_premption(FAR struct note_driver_s *drv,
+static void notelog_preemption(FAR struct note_driver_s *drv,
                               FAR struct tcb_s *tcb, bool locked)
```
Comment (xiaoxiang781216) [inline drivers/note/notelog_driver.c:259]:
```suggestion
                               FAR struct tcb_s *tcb, bool locked)
```

File: drivers/note/notesnap_driver.c [hunk 4]
```diff
@@ -86,7 +86,7 @@ static void notesnap_cpu_resumed(FAR struct note_driver_s *drv,
 #  endif
 #endif
 #ifdef CONFIG_SCHED_INSTRUMENTATION_PREEMPTION
-static void notesnap_premption(FAR struct note_driver_s *drv,
+static void notesnap_preemption(FAR struct note_driver_s *drv,
                                FAR struct tcb_s *tcb, bool locked);
```
Comment (xiaoxiang781216) [inline drivers/note/notesnap_driver.c:90]:
```suggestion
                                FAR struct tcb_s *tcb, bool locked);
```

File: drivers/note/notesnap_driver.c [hunk 5]
```diff
@@ -302,7 +302,7 @@ static void notesnap_cpu_resumed(FAR struct note_driver_s *drv,
 #endif
 
 #ifdef CONFIG_SCHED_INSTRUMENTATION_PREEMPTION
-static void notesnap_premption(FAR struct note_driver_s *drv,
+static void notesnap_preemption(FAR struct note_driver_s *drv,
                                FAR struct tcb_s *tcb, bool locked)
```
Comment (xiaoxiang781216) [inline drivers/note/notesnap_driver.c:306]:
```suggestion
                                FAR struct tcb_s *tcb, bool locked)
```

File: include/nuttx/sched.h [hunk 6]
```diff
@@ -702,10 +702,10 @@ struct tcb_s
 #endif
 
 #if CONFIG_SCHED_CRITMONITOR_MAXTIME_PREEMPTION >= 0
-  clock_t premp_start;                   /* Time when preemption disabled   */
-  clock_t premp_max;                     /* Max time preemption disabled    */
-  void   *premp_caller;                  /* Caller of preemption disabled   */
-  void   *premp_max_caller;              /* Caller of max preemption        */
+  clock_t preemp_start;                   /* Time when preemption disabled   */
```
Comment (xiaoxiang781216) [inline include/nuttx/sched.h:705]:
```suggestion
  clock_t preemp_start;                  /* Time when preemption disabled   */
```

**Pull request merged**